lib/chef-dk/policyfile_compiler.rb

```diff
@@ -123,7 +126,11 @@ def install
     end
 
     def create_spec_for_cookbook(cookbook_name, version)
-      source_options = default_source.source_options_for(cookbook_name, version)
+      matching_source = default_source.find { |s|
```

Comment:
I thought so from looking at the code, but you can only get here via `#install` after `#graph_solution` is called. There's no way to set up a Policyfile so that `#graph_solution` doesn't fail and also there is no source for the cookbook. In any case, I added some tests to demonstrate this.